libraries/bsg_manycore_request_packet.h

```diff
@@ -58,19 +58,65 @@ extern "C" {
                 uint8_t  y_dst; //!< y coordinate of the responder
                 uint8_t  x_src; //!< x coordinate of the requester
                 uint8_t  y_src; //!< y coordinate of the requester
-                uint32_t data;  //!< packet's payload data
+                /*
+                  match op_v2 with
+                  [load]     => load_info
+                  [sw,amo_*] => payload data
+                  [store]    => payload_data + reg_id in unmasked bytes
+                 */
+                uint32_t payload;  //!< packet's payload data
+                /*
+                  match op_v2 with
+                  [store]      => store_mask
+                  [sw,load]    => register_id
+                  [cache_op]   => afl,ainv,aflinv,tagfl
+                 */
                 uint8_t  reg_id; //!< 5-bit id for load or amo
-                uint8_t  op_ex;  //!< 4-bit byte mask
-                uint8_t  op;    //!< opcode
+                uint8_t  op_v2;    //!< opcode
                 uint32_t addr;  //!< address field (EPA)
                 uint8_t  reserved[1];
         }  __attribute__((packed)) hb_mc_request_packet_t;
 
+        typedef struct hb_mc_request_packet_load_info {
+                uint32_t part_sel;
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_PARTSEL_MASK         0x60
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_PARTSEL_SHIFT        5
+                int      is_hex_op;
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_IS_HEX_OP_MASK       0x10
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_IS_HEX_OP_SHIFT      4
+                int      is_byte_op;
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_IS_BYTE_OP_MASK      0x08
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_IS_BYTE_OP_SHIFT     3
+                int      is_unsigned_op;
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_IS_UNSIGNED_OP_MASK  0x04
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_IS_UNSIGNED_OP_SHIFT 2
+        } hb_mc_request_packet_load_info_t;
+
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_GET(data, field)                 \
+                (((data) & HB_MC_REQUEST_PACKET_LOAD_INFO_ ## field ##_MASK) >> \
+                 HB_MC_REQUEST_PACKET_LOAD_INFO_ ## field ## _SHIFT)
+
+#define HB_MC_REQUEST_PACKET_LOAD_INFO_SET(data, field, field_val)      \
+        do {                                                            \
+                data &= ~HB_MC_REQUEST_PACKET_LOAD_INFO_ ## field ## _MASK; \
+                data |= ((field_val) << HB_MC_REQUEST_PACKET_LOAD_INFO_ ## field ## _SHIFT) & \
+                        HB_MC_REQUEST_PACKET_LOAD_INFO_ ## field ## _MASK; \
```

Comment:
Is this what you meant when you said you went whole hog on macros?

Reply:
Less gross than spaghetti shift and anding.